bibcat/core/paper.py

```diff
@@ -255,11 +263,134 @@ def _check_truematch(self, text, keyword_objs, dict_ambigs, do_verbose=None, do_
                - text_wordchunk: the chunk of text searched
                - text_database: the phrase searched within the chunk of text
                - matcher: the result of regex search
-               - ind: index number within the database
-               - exp: ambiguous phrases searched
+        Raises:
+        - ValueError: If no wordchunks are identified after short-circuit checks,
+            raises an error with detailed part-of-speech diagnostics.
+        - NotImplementedError: If a wordchunk was found after short-circuit checks,
+            but no matches or meanings are found in the ambiguous database for the wordchunk.
+        """
+
+        # Set up initial variables
+        setup_data = self._setup_check_truematch_vars(text, dict_ambigs, keyword_objs, do_verbose, do_verbose_deep)
+
+        # Short-circuit checks
+        for check in (
+            self._early_true_non_ambig_keywords,
+            self._early_false_no_keyword_match,
+            self._early_true_acronym_match,
+            self._early_true_non_ambig_phrases,
+        ):
+            result = check(setup_data)
+            if result is not None:
+                return result
+
+        # Assemble makeshift wordchunks (not using NLP ones here)
+        # Not sure why happened, but NLP sometimes failed to identify nouns/num.
+        list_wordchunks = self._assemble_keyword_wordchunks_wrapper(setup_data)
+
+        # Short-circuit check for exact wordchunks
+        result = self._early_true_exact_wordchunk(list_wordchunks, setup_data)
+        if result is not None:
+            return result
 
+        # Iterate through wordchunks to determine true vs false match status
+        list_results = [self._consider_wordchunk(curr_chunk, setup_data) for curr_chunk in list_wordchunks]
+
+        # Combine the results and return overall boolean match
+        fin_result = {
+            "bool": any([(item["bool"]) for item in list_results]),
+            "info": [item["info"][0] for item in list_results],
+        }
+        return fin_result
+
+    @dataclass
+    class TruematchSetup:
+        """
```

Comment:
this is a class defined within a class?  Since this is setup, should this be moved to outside of `Paper`?